src/functions/interaction/clear.js

```diff
@@ -1,56 +1,78 @@
 module.exports = async (d) => {
-    const {code} = d.command;
-    const inside = d.unpack();
-    const err = d.inside(inside);
-    if (err) return d.error(err);
-
-    let [amt, filter = "everyone", returnCount = "false", channelID = d.channel.id] =
-        inside.splits;
-
-    amt = Number(amt);
-    if (isNaN(amt))
-        return d.aoiError.fnError(
-            d,
-            "custom",
-            {inside},
-            "Invalid Amout Provided In",
-        );
-
-    const channel = await d.util.getChannel(d, channelID);
-    if (!channel) return d.aoiError.fnError(d, "channel", {inside});
-
-    let messages = await channel.messages
-        .fetch({limit: 100, cache: false})
-        .catch((err) => {
-            d.aoiError.fnError(
-                d,
-                "custom",
-                {},
-                "Failed To Fetch Messages With Reason: " + err,
-            );
-        });
-
-    messages = [...messages.filter((x) =>
-        filter === "everyone"
-            ? true
-            : filter === "unPins"
-                ? !x.pinned
-                : filter === "bot"
-                    ? x.author?.bot
-                    : x.author?.id === filter,
-    ).values()].slice(0, amt);
-
-    let result = await channel.bulkDelete(messages, true).catch((err) => {
-        d.aoiError.fnError(
-            d,
-            "custom",
-            {},
-            "Failed To Delete Message With Reason: " + err,
-        );
+  const data = d.util.aoiFunc(d);
+  const { code } = d.command;
+  if (data.err) return d.error(data.err);
+
+  let [
+    channelID = d.channel.id,
+    amount,
+    filters = "everyone",
+    returnCount = false,
+  ] = data.inside.splits;
+
+  if (isNaN(amount))
+    return d.aoiError.fnError(
+      d,
+      "custom",
+      { inside: data.inside },
+      "Amount Provided In"
+    );
+  amount = +amount + 1;
+
+  const channel = await d.util.getChannel(d, channelID);
+  if (!channel)
+    return d.aoiError.fnError(d, "channel", { inside: data.inside });
+
+  let messages = await channel.messages
+    .fetch({ limit: 100, cache: false })
+    .catch((err) => {
+      d.aoiError.fnError(
+        d,
+        "custom",
+        {},
+        "Failed To Fetch Messages With Reason: " + err
+      );
     });
-    result = returnCount === "true" ? result.size : undefined;
 
-    return {
-        code: d.util.setCode({function: d.func, code, inside, result}),
-    };
-};
\ No newline at end of file
+  filters = filters.toLowerCase().split(",");
+
+  messages = [...messages.values()]
+    .filter((x) => {
+      if (filters.includes("everyone")) return true;
+      if (filters.includes("notpinned") && x.pinned) return false;
+      if (filters.includes("bots") && x.author?.bot) return true;
+      if (
+        filters.some(
+          (filter) =>
+            filter.startsWith("user:") && x.author?.id === filter.split(":")[1]
```

Comment:
Unsure about this one, if it's needed or not. However, using this slows the response down around 2-4 seconds depending on amount.